src/org/zaproxy/zap/extension/alertFilters/DialogAddAlertFilter.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.alertFilters;
+
+import java.awt.Dialog;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+
+import javax.swing.JCheckBox;
+import javax.swing.JComboBox;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.zaproxy.zap.model.Context;
+import org.zaproxy.zap.utils.ZapTextField;
+import org.zaproxy.zap.view.AbstractFormDialog;
+import org.zaproxy.zap.view.LayoutHelper;
+
+/**
+ * The Dialog for adding and configuring a new {@link AlertFilter}.
+ */
+public class DialogAddAlertFilter extends AbstractFormDialog {
+
+	/** The Constant serialVersionUID. */
+	private static final long serialVersionUID = -7210879426146833234L;
+
+	/** The Constant logger. */
+	protected static final Logger log = Logger.getLogger(DialogAddAlertFilter.class);
+
+	private static final String DIALOG_TITLE = Constant.messages.getString("alertFilters.dialog.add.title");
+	private static final String CONFIRM_BUTTON_LABEL = 
+			Constant.messages.getString("alertFilters.dialog.add.button.confirm");
+
+	//private ExtensionAlertFilters extension = null;
+	private JPanel fieldsPanel;
+	private JCheckBox enabledCheckBox;
+	private JComboBox<String> alertCombo;
+	private JComboBox<String> newLevelCombo;
+	private ZapTextField urlTextField;
+	private JCheckBox regexCheckBox;
+	private ZapTextField paramTextField;
+	protected Context workingContext;
+	protected AlertFilter alertFilter;
+
+	/**
+	 * Instantiates a new dialog add alertFilter.
+	 * 
+	 * @param owner the owner
+	 * @param extension the extension
+	 * @param context the context
+	 */
+	public DialogAddAlertFilter(Dialog owner, ExtensionAlertFilters extension) {
```

Comment:
The "extension" parameter could be removed, it's not used.
It could also lead to the removal of the "extension" parameter in ContextAlertFilterPanel and AlertFiltersMultipleOptionsPanel.